src/instructlab/model/train.py

```diff
@@ -520,20 +580,340 @@ def get_files(directory: str, pattern: str) -> list[str]:
+    model_override: typing.Optional[str] = None,
+    ckpt_dir: typing.Optional[str] = None,
+) -> None:
+    """A single step of end-to-end training."""
+
+    from instructlab.training import (  # pylint: disable=no-name-in-module
+        run_training,
+    )
+
+    # override whatever got injected at a higher level
+    logger.debug(
+        f"E2E -- training phase -- Overriding data_path: {train_args.data_path} with {data_path}"
+    )
+    train_args.data_path = data_path
+
+    if ckpt_dir:
+        train_args.ckpt_output_dir = ckpt_dir
+
+    if model_override:
+        logger.debug(
+            f"E2E -- training phase -- Overriding model_path: {train_args.model_path} with {model_override}"
+        )
+        train_args.model_path = model_override
+
+    logger.debug(
+        f"E2E -- training phase -- Current training args for phase: {pp.pformat(train_args)}"
+    )
+
+    try:
+        run_training(train_args=train_args, torch_args=torch_args)
+    except Exception as e:
+        logging.fatal(e)
+        sys.exit(1)
+
+
+def _mmlu(ckpt_path: str) -> float:
+    from instructlab.eval.mmlu import MMLUEvaluator, MMLU_TASKS
+    import torch
+
+    evaluator = MMLUEvaluator(ckpt_path, tasks=MMLU_TASKS, few_shots=5)
+
+    ckpt_score, _ = evaluator.run()
+
+    logging.debug("E2E -- MMLU eval phase -- Clearing PyTorch cache")
+    torch.cuda.empty_cache()
+
+    return ckpt_score
+
+
+def _mtbench(ctx, ckpt_path: Path, eval_cache_path: str, judge_name: str) -> float:
+    # TODO: run all generations in serial and then do all judgements at once to save time loading/unloading prometheus.
+
+    # dynamic import- eval code not used outside of this path.
+    from instructlab.eval.mt_bench import MTBenchEvaluator
+
+    from instructlab.model.evaluate import launch_server
+    import torch
+
+    ckpt_path = str(ckpt_path)
+    eval_cache_path = str(eval_cache_path)
+
+    # hard-override for local insecure vLLM serving
+    ctx.params["tls_client_cert"] = None
+    ctx.params["tls_client_key"] = None
+    ctx.params["tls_client_passwd"] = None
+    ctx.params["tls_insecure"] = True
+
+    evaluator = MTBenchEvaluator(
+        model_name=ckpt_path,
+        judge_model_name=judge_name,
+        output_dir=eval_cache_path,
+        merge_system_user_message=True,
+    )
+
+    server = None
+    model_serve_url = None
+    try:
+        logger.debug("Starting model server for mt-bench answer generation")
+        server, model_serve_url = launch_server(
+            ctx=ctx,
+            model=ckpt_path,
+            model_name=ckpt_path,
+            gpus=torch.cuda.device_count(),
+            max_workers=None,  # not needed for vLLM
+            enable_serving_output=False,
+            backend="vllm",
+        )
+        logger.debug("Generating mt-bench answers")
+        evaluator.gen_answers(model_serve_url)
+    finally:
+        if server is not None:
+            server.shutdown()
+
+    ckpt_score: typing.Optional[int] = None
```

Comment:
Is this type correct or is it the one in the fuction signature?

Reply:
ack